pkg/harvester/edit/harvesterhci.io.host/HarvesterSeeder.vue

```diff
@@ -0,0 +1,306 @@
+
+const _NEW = '_NEW';
+
+export default {
+  name: 'HarvesterSeeder',
+
+  components: {
+    Checkbox,
+    LabeledInput,
+    LabeledSelect,
+    RadioGroup,
+    ModalWithCard,
+    NameNsDescription,
+  },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true
+    },
+
+    node: {
+      type:     Object,
+      required: true,
+    },
+
+    registerBeforeHook: {
+      type:     Function,
+      required: true,
+    },
+
+    inventory: {
+      type:     Object,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      enabled:           true,
+      value:             this.inventory,
+      secret:            {},
+      errors:            [],
+      newSecretSelected: false,
+    };
+  },
+
+  created() {
+    this.registerBeforeHook(this.saveInventory, 'saveInventory');
+  },
+
+  async fetch() {
+    const inStore = this.$store.getters['currentProduct'].inStore;
+
+    this.secret = await this.$store.dispatch(`${ inStore }/create`, {
+      type: SECRET,
+      data: {
+        username: '',
+        password: '',
+      },
+      metadata: {
+        namespace: '',
+        name:      '',
+        describe:  '',
+      },
+    });
+  },
+
+  computed: {
+    secretOption() {
+      const inStore = this.$store.getters['currentProduct'].inStore;
+
+      const out = this.$store.getters[`${ inStore }/all`](SECRET).filter((s) => {
+        return s.data?.username && s.data?.password;
+      }).map( (s) => {
+        return {
+          label: s.id,
+          value: s.id
+        };
+      });
+
+      // if (!(this.disableCreate || this.mode === _VIEW) && this.isCreatable) {
+      out.unshift({
+        label: this.t('harvester.virtualMachine.createSSHKey'),
+        value: _NEW,
+      });
+      // }
+
+      return out;
+    },
+
+    selectedSecret: {
+      get() {
+        const namespace = this.value.spec?.baseboardSpec?.connection?.authSecretRef?.namespace;
+        const name = this.value?.spec?.baseboardSpec?.connection?.authSecretRef?.name;
+
+        if (namespace && name) {
+          return `${ namespace }/${ name }`;
+        } else {
+          return '';
+        }
+      },
+
+      set(value) {
+        if (value === _NEW) {
+          this.newSecretSelected = true;
+        } else {
+          const [namespace, name] = value.split('/');
+
+          this.$set(this.value.spec.baseboardSpec.connection.authSecretRef, 'namespace', namespace);
+          this.$set(this.value.spec.baseboardSpec.connection.authSecretRef, 'name', name);
+        }
+      },
+    },
+
+    username: {
+      get() {
+        return base64Decode(this.secret?.data?.username);
+      },
+
+      set(value) {
+        this.$set(this.secret.data, 'username', base64Encode(value));
+      }
+    },
+
+    password: {
+      get() {
+        return base64Decode(this.secret?.data?.password);
+      },
+
+      set(value) {
+        this.$set(this.secret.data, 'password', base64Encode(value));
+      }
+    },
```

Comment:
It might be more appropriate not to use the computed attribute。

Reply:
Could you explain the reason to me? I think this is a universal approach.

![image](https://user-images.githubusercontent.com/18737885/231941885-7fd4916e-8d40-4e6e-b5e3-fc79e5d76ffa.png)